lib/src/connector/score_connector.dart

```diff
@@ -141,13 +135,14 @@ class ScoreConnector {
       result = await Connector.getDataByGet(parameter);
       tagNode = parse(result);
       rankNodes = tagNode.getElementsByTagName("tbody")[0].getElementsByTagName("tr");
-      rankNodes = rankNodes.getRange(2, rankNodes.length).toList().reversed.toList();
+      final filteredRankNodes =
+          rankNodes.where((node) => node.getElementsByTagName("td").length >= 7).toList().reversed.toList();
+      rankNodes = filteredRankNodes;
```

Comment:
yes it seems a good way to create filtered rank nodes, but we should create a handler when tableContent is empty

Reply:
Em... But I haven't seen any handler for this case in the file.
Do you mean, to create that handler in the future ??

Reply:
It seems that the previous implementation was based on not specifically handling empty `rankNodes`.

Reply:
originally, the program would catch the error at parsing the double value from the invalid table, though it is not intended

Reply:
but they are in different context, so i think it is fine for now

Reply:
```dart
      final rankNodes = tagNode
          .getElementsByTagName("tbody")
          .first
          .getElementsByTagName("tr")
          .reversed
          .where((row) => row.getElementsByTagName("td").length >= 7)
          .toList(growable: false);
```

If we expect it raises an error when nothing inside `<tbody>`, how about just ignore the `isNotEmpty` check ...

Reply:
yes i think it will have a same result

Reply:
I think these two different approaches should depends on the test.
Perhaps we may try to see whether this error is necessary to be risen.

Reply:
ok

Reply:
should i commit this?

Reply:
For now we know, if that error rose, the final return value from the whole function would be `null`
However, if we not let it rise that, we may have a "higher chance" to get an empty list instead of `null`. (There are still other places could throw error)

The only usecase of `ScoreConnector.getScoreRankList` is at the `score_rank_task.dart`.

```dart
    if (status == TaskStatus.success) {
      super.onStart(R.current.getScoreRank);
      final value = await ScoreConnector.getScoreRankList() as List<SemesterCourseScoreJson>?;
      super.onEnd();
      if (value != null) {
        result = value;
        return TaskStatus.success;
      } else {
        return super.onError(R.current.getScoreRankError);
      }
    }
```

So here we may go to check what will happened when `value` no longer to be `null`.

Reply:
> should i commit this?

It depends on you.
Since I haven't build this branch to see what exact behavior is, you may find your answer by yourself XDD.

Reply:
usually when I get an error on getting information on TAT, i will just cancel it and reload asap, so i think it really depends on the test.
but checking what happened when the value is not null but also not valid is important, it will make us solve the issue easier when user provides error code or screen shots